aws-lc-rs/src/ec.rs

```diff
@@ -259,7 +264,29 @@ pub(crate) unsafe fn ec_key_from_private(
     if 1 != EC_KEY_set_public_key(*ec_key, *pub_key) {
         return Err(Unspecified);
     }
+    // Validate the EC_KEY before returning it.
+    let bits: c_uint = EC_GROUP_order_bits(**ec_group).try_into()?;
+    validate_ec_key(&ec_key.as_const(), bits)?;
+
+    Ok(ec_key)
+}
 
+#[inline]
+pub(crate) unsafe fn ec_key_generate(
+    ec_group: &ConstPointer<EC_GROUP>,
+) -> Result<DetachableLcPtr<*mut EC_KEY>, Unspecified> {
+    let ec_key = DetachableLcPtr::new(EC_KEY_new())?;
+    if 1 != EC_KEY_set_group(*ec_key, **ec_group) {
+        return Err(Unspecified);
+    }
```

Comment:
I suggest using `EC_KEY_new_by_curve_name` instead of `EC_KEY_new()` + `EC_KEY_set_group`.

Reply:
Fixed in the next commit.